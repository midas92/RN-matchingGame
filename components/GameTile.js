import React from 'react';
import FlippableCard from './FlippableCard';

const GameTile = ({
  rowNumber,
  columnNumber,
  value,
  selected,
  locked,
  onTilePressed,
  totalColumns
}) => {

  const handleTilePressed = () => {
    onTilePressed({
      rowNumber: rowNumber,
      columnNumber: columnNumber,
      value: value,
      selected: selected,
      locked: locked 
    })
  }

  const tileStyle = {
    height: 300 / totalColumns,
    width: 300 / totalColumns,
    borderRadius: '10%',

    alignItems: 'center',
    justifyContent: 'center',
    margin: 3,
    shadowColor: 'black',
    shadowRadius: 2,
    shadowOpacity: 1,
  }

  return (
    <FlippableCard
      value={value}
      onTilePressed={handleTilePressed}
      totalColumns={totalColumns}
      cardStyle={tileStyle}
      flipped={locked || selected }
    />
  );
}

export default GameTile;
