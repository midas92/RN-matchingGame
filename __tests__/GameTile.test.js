import React from 'react';
import { render } from '@testing-library/react-native';
import GameTile from '../components/GameTile';

describe('<GameTile/>', () => {
  it('should match snapshot', () => {
    const result = render(<GameTile/>).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('should render the GameTile component', () => {
    const result = render(<GameTile/>);
    expect(result).toBeTruthy();
  });
});
