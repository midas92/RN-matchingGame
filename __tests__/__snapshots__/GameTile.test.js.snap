// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameTile/> should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backfaceVisibility": "hidden",
        "backgroundColor": "gray",
        "borderRadius": "10%",
        "height": NaN,
        "justifyContent": "center",
        "margin": 3,
        "shadowColor": "black",
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "transform": Array [
          Object {
            "rotateX": "0deg",
          },
        ],
        "width": NaN,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backfaceVisibility": "hidden",
        "backgroundColor": "#34a8eb",
        "borderRadius": "10%",
        "height": NaN,
        "justifyContent": "center",
        "margin": 3,
        "position": "absolute",
        "shadowColor": "black",
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "transform": Array [
          Object {
            "rotateX": "180deg",
          },
        ],
        "width": NaN,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "borderRadius": "10%",
            "height": NaN,
            "justifyContent": "center",
            "margin": 3,
            "shadowColor": "black",
            "shadowOpacity": 1,
            "shadowRadius": 2,
            "width": NaN,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": NaN,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
